### ISSUE TITLE
harf: bad rendering with \prehyphenchar=0

### ISSUE THREAD

**Issue by jbezos** [latex3/luaotfload#207]
Consider the following example:
```tex
\documentclass{article}

\usepackage{luahyphenrules} % A short package to load patterns
\HyphenRules{bengali}

\usepackage{fontspec}
\setmainfont{fontkalpurush.ttf}[Renderer=Harfbuzz, Script=Bengali]

\patterns{f3i}

\begin{document}

নেস্টেড

\prehyphenchar=0

নেস্টেড

\end{document}
```
![gap-prehyphenchar](https://user-images.githubusercontent.com/1314536/144829768-07386aa4-b0fa-41cd-b61d-1a471eebe11b.jpg)

The second one is not correctly rendered because there is gap. Actually, Bengali doesn’t require this setting, and very likely scripts requiring it are not affected (I’m not sure), but clearly there is something wrong.

**Comment by zauguin** [comment]
I've got a fix for this, but I want to fix our tests before committing it.

**Comment by zauguin** [comment]
Should be fixed in 821c98b3da0369dea2f2bb2a509f1ec28f49998d.

**Comment by zauguin** [comment]
Fixed in the latest release.
